game-core/src/main/java/games/strategy/engine/data/unit/ability/CombatUnitAbility.java

```diff
@@ -0,0 +1,114 @@
+package games.strategy.engine.data.unit.ability;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.NonFinal;
+
+/** Unit Ability for combat situations */
+@Data
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Builder(toBuilder = true)
+public class CombatUnitAbility {
+
+  public static final CombatUnitAbility EMPTY = CombatUnitAbility.builder().name("Empty").build();
+
+  enum DiceType {
+    // Use AA dice (attackAA, offensiveAttackAA)
+    AA,
+    // Use bombard dice (bombard or attack if bombard is empty)
+    BOMBARD,
+    // Use normal dice (attack, defense)
+    NORMAL
+  }
+
+  /** The name of this unit ability that will be shown in the Battle UI */
+  @NonNull String name;
+
+  /** The unit types that have this ability */
+  @Builder.Default @NonFinal Collection<UnitType> attachedUnitTypes = List.of();
+
+  /** The unit types that can be targeted */
+  @Builder.Default Collection<UnitType> targets = List.of();
+
+  /** The type of dice that will be rolled. */
+  @Builder.Default DiceType diceType = DiceType.NORMAL;
+
+  /** The side(s) that the unit needs to be on to use the ability */
+  @Builder.Default
+  Collection<BattleState.Side> sides = List.of(BattleState.Side.OFFENSE, BattleState.Side.DEFENSE);
+
+  /** How many combat rounds this ability can be used */
+  @Builder.Default int round = Integer.MAX_VALUE;
+
+  /**
+   * Can the casualties return fire after the battle phase that this UnitAbility is active?
+   *
+   * <p>Casualties that are created in the battle phase where this UnitAbility is active can always
+   * return fire since they are firing simultaneously
+   */
+  @Builder.Default boolean returnFire = true;
+
+  /**
+   * Does this unit commit suicide after it has fired a shot?
+   *
+   * <p>Whether the shot hit or not doesn't matter.
+   *
+   * <p>It can commit suicide as a defensive unit, offensive unit, or both.
+   */
+  @Builder.Default Collection<BattleState.Side> commitSuicide = List.of();
```

Comment:
I was thinking about this one, the commit suicide on offense, defense, or after a hit or any roll might make for a good enum to flag which cases.

Regardless, the variable name here sounds more like a boolean rather than a list. I'm not sure if there is a better name, but could be worth a second look.

Reply:
I've renamed them and added an enum.  Take a look at https://github.com/triplea-game/triplea/pull/8646/commits/4fe6eda8477965dbf1c05ef4901f35e18dce612d